ide/libs.tomljava/manifest.mf

```diff
@@ -0,0 +1,4 @@
+Manifest-Version: 1.0
+OpenIDE-Module: org.netbeans.libs.tomljava/3
+OpenIDE-Module-Localizing-Bundle: org/netbeans/libs/tomljava/Bundle.properties
+OpenIDE-Module-Specification-Version: 1.0
```

Comment:
consider adding:
```
OpenIDE-Module-Java-Dependencies: Java > 11
```
This would prevent the module and the dependency tree to be loaded on platform apps using older JDKs. Otherwise it will fail later. (I don't think the module system looks at source/target)

Reply:
So that should say `OpenIDE-Module-Java-Depencencies: Java >= 11`, right? Worth a change?

Reply:
its confusing but `>` means `>=` 

we have this everywhere https://github.com/search?q=repo%3Aapache%2Fnetbeans%20OpenIDE-Module-Java-Dependencies%3A%20Java%20%3E%2011&type=code

Reply:
It also should be failing a test if it's not there?!

Reply:
not sure if the test takes lib wrappers into account, it might only look at class files of the project file?

Reply:
AFAIU library wrapper class versions are not taken into account in the current `ant commit-validation` tests.

Reply:
No, sorry, misread which manifest this was.

Reply:
Actually, looking back at #4775 and my change before that, I think libraries are meant to be checked too. It used to be every JAR. Is this definitely compiled for release 11?

Reply:
antlr4-runtime-4.13.1 is still Java 8 (thought it was 11), toml-java is Java 11.

```sh
$ javap -v -cp antlr4-runtime-4.13.1.jar org.antlr.v4.runtime.CharStream
Classfile jar:file:///[...]/netbeans/ide/libs.antlr4.runtime/external/antlr4-runtime-4.13.1.jar!/org/antlr/v4/runtime/CharStream.class
  Last modified 4 sept 2023; size 237 bytes
  SHA-256 checksum f56df94dc6d8100a59ee452da9e9cdd30b94523ce8e297daeab781193aed81c1
  Compiled from "CharStream.java"
public interface org.antlr.v4.runtime.CharStream extends org.antlr.v4.runtime.IntStream
  minor version: 0
  major version: 52
```

Reply:
OK, the test should be picking up a problem here as far as I can tell.  I think it might have become too lenient!  I think there's an issue stemming from #4775 where it's ignoring modules without a declared Java dependency, and I think also ignoring non-referenced autoloads.  The test looks like it potentially needs `honorAutoloadEager` switched off, and to always add a default version for all module referenced JARs in https://github.com/apache/netbeans/blob/master/platform/o.n.core/test/qa-functional/src/org/netbeans/core/validation/ValidateClassFilesTest.java#L129

Interestingly, also noticed that there is code to add Java dependency automatically for value of `javac.target` that has a now incorrect regex protecting it?? https://github.com/apache/netbeans/blob/master/nbbuild/antsrc/org/netbeans/nbbuild/JarWithModuleAttributes.java#L221  (I was looking for where this dependency was being added automatically when I noticed the test now relies on its presence).

Reply:
regarding JarWithModuleAttributes.java: we should check `javac.release` there too. However the regexp looks good to me, no? I didn't know that this existed - this essentially makes all the manually added manifest fields redundant, no?

Reply:
I assumed from the conversation above that the built module manifest didn't have the version in it.  Checking the dev build shows this attribute is included already.  I misread the regexp as if using `+` rather than `*` because I thought it wasn't passing the if check. All in all a long way around to no change required here then? :smile:

The test does still need looking at, but should be passing for the right reasons here.

And, yes, explicitly added manifest fields should be redundant for cases where `javac.target` is set then.  There are a few places this is used where this wasn't the case (eg. JShell support IIRC).

Reply:
In fact I can easily downgrade `toml-java` to JDK8 binary format for the time being. Let's do that.

Reply:
@vieiro no wait, no downgrade is needed - (almost) everything is working as intended.

Reply:
D'oh! I've downgraded to JDK8. No worries, we can always upgrade to JDK11.
After all the objective of `toml-java` was for it to be reusable on other projects, the lower the JDK the more projects can use it.

Reply:
@neilcsmith-net scraped a PR together from the things we discovered in this discussion: https://github.com/apache/netbeans/pull/7188